cpp/src/arrow/compute/kernels/vector_sort_test.cc

```diff
@@ -243,25 +258,23 @@ TYPED_TEST(TestSortToIndicesKernelForReal, SortReal) {
   this->AssertSortToIndices("[]", "[]");
 
   this->AssertSortToIndices("[3.4, 2.6, 6.3]", "[1, 0, 2]");
-
   this->AssertSortToIndices("[1.1, 2.4, 3.5, 4.3, 5.1, 6.8, 7.3]", "[0,1,2,3,4,5,6]");
-
   this->AssertSortToIndices("[7, 6, 5, 4, 3, 2, 1]", "[6,5,4,3,2,1,0]");
-
   this->AssertSortToIndices("[10.4, 12, 4.2, 50, 50.3, 32, 11]", "[2,0,6,1,5,3,4]");
 
   this->AssertSortToIndices("[null, 1, 3.3, null, 2, 5.3]", "[1,4,2,5,0,3]");
+
+  this->AssertSortToIndices("[3, 4, NaN, 1, 2, null]", "[3,4,0,1,2,5]");
+  this->AssertSortToIndices("[NaN, 2, NaN, 3, 1]", "[4,1,3,0,2]");
+  this->AssertSortToIndices("[null, NaN, NaN, null]", "[1,2,0,3]");
```

Comment:
Since the sort is non-stable, the result could also be `[2,1,0,3]`, etc., no?

Reply:
Sort is stable in current implementation.